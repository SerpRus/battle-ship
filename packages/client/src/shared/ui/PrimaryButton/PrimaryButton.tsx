import React from 'react';
import { Button } from 'antd';

const PrimaryButton: React.FC<{ buttonTitle: string }> = ({ buttonTitle }) => (
  <Button type="primary" htmlType="submit">
    {buttonTitle}
  </Button>
);

export default PrimaryButton;
