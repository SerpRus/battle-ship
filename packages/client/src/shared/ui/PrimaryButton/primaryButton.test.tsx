// import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import PrimaryButton from './PrimaryButton';

describe('PrimaryButton', () => {
  test('should render button with correct title', () => {
    const buttonTitle = 'Submit';
    render(<PrimaryButton>{buttonTitle}</PrimaryButton>);

    const buttonElement = screen.getByText(buttonTitle);
    expect(buttonElement).toBeInTheDocument();
  });

  // TODO: commented cause differences in classnames in CI and local
  /* it('should match snapshot', () => {
    const buttonTitle = 'hello!';
    const tree = renderer
      .create(<PrimaryButton>{buttonTitle}</PrimaryButton>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  }); */

  it('should render correctly title', () => {
    const buttonTitle = 'Bye-bye!!!';
    const { getByText } = render(<PrimaryButton>{buttonTitle}</PrimaryButton>);

    const buttonTitleTest = getByText('Bye-bye!!!');
    expect(buttonTitleTest).toBeInTheDocument();
  });
});
