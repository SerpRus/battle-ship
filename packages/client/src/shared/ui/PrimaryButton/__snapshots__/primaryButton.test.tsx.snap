// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton should match snapshot 1`] = `
<button
  className="ant-btn css-dev-only-do-not-override-2i2tap ant-btn-primary"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  type="submit"
>
  <span>
    hello!
  </span>
</button>
`;
